service/discovery/azure/storage.go

```diff
@@ -150,7 +151,13 @@ func (d *azureStorageDiscovery) discoverCosmosDB() ([]voc.IsCloudResource, error
 			return res.Value
 		},
 		func(dbAccount *armcosmos.DatabaseAccountGetResults) error {
-			cosmos, err := d.handleCosmosDB(dbAccount)
+			// Get ActivityLogging for the CosmosDB
+			activityLogging, err := d.discoverDiagnosticSettings(util.Deref(dbAccount.ID))
```

Comment:
maybe move that part to the `handleCosmosDB` function?

Reply:
I would prefer to add all calls to Azure to the discoverX() functions. HandleX() functions schould not call Azure endpoints, but only process the information from Azure. I know we haven't done this in the past, but we should.